File: Task1/Task1.rb
def print_pattern(n)
  ch = 'A'.ord + n - 1
  (n).downto(1) do |i|
    (0..(n - i)).each do |j|
      print "#{(ch - j).chr} "
    end
    puts
  end
end

n = 0
print "Введите значение N (положительное число): "
while true
  input = gets.chomp
  if input =~ /^\d+$/ && (n = input.to_i) > 0
    break
  end
  print "Введите корректное положительное значение: "
end
print_pattern(n)

File: Task1/Task1.py
def print_pattern(N):
    ch = chr(ord('A') + N - 1)
    for i in range(N, 0, -1):
        for j in range(N - i + 1):
            print(chr(ord(ch) - j), end=' ')
        print()

N = 0
while True:
    try:
        N = int(input("Введите значение N (положительное число): "))
        if N > 0 and N <= 26:
            break
    except ValueError:
        pass
print_pattern(N)
